src/update-feeds/check-feeds.ts

```diff
@@ -0,0 +1,164 @@
+  return (
+    batch
+      .map((dapi) => ({
+        ...dapi,
+        decodedDataFeed: {
+          ...dapi.decodedDataFeed,
+          beacons: dapi.decodedDataFeed.beacons.map((beacon) => {
+            const onChainValue = onChainFeedValues.find((onChainValue) => beacon.beaconId === onChainValue.beaconId)
+              ?.onChainValue ?? {
+              value: ethers.BigNumber.from(1),
+              timestamp: ethers.BigNumber.from(1),
+            };
+            const signedData = getStoreDataPoint(beacon.beaconId) ?? {
+              timestamp: '1',
+              airnode: ethers.constants.AddressZero,
+              encodedValue: ethers.constants.HashZero,
+              templateId: ethers.constants.HashZero,
+              signature: ethers.constants.HashZero,
+            };
+            const offChainValue = {
+              timestamp: ethers.BigNumber.from(signedData.timestamp),
+              value: decodeBeaconValue(signedData.encodedValue)!,
+            };
+
+            const localDataIsStale =
+              signedData?.airnode === ethers.constants.AddressZero ||
+              offChainValue.timestamp.lt(onChainValue.timestamp) ||
+              offChainValue.timestamp.gt(Math.ceil(Date.now() / 1000 + 60 * 60));
+            const currentValue = offChainValue.timestamp.gt(onChainValue.timestamp) ? offChainValue : onChainValue;
+
+            return {
+              ...beacon,
+              currentValue,
+              localDataIsStale,
+              signedData,
+            };
+          }),
+        },
+      }))
+      // Filter out dapis that cannot be updated
+      .filter((dapi) => {
+        const newBeaconSetValue = calculateMedian(
+          dapi.decodedDataFeed.beacons.map((beacon) => beacon.currentValue.value)
+        );
+        const newBeaconSetTimestamp = calculateMedian(
+          dapi.decodedDataFeed.beacons.map((beacon) => beacon.currentValue.timestamp)
+        )!.toNumber();
+        const adjustedDeviationThresholdCoefficient = multiplyBigNumber(
+          dapi.updateParameters.deviationThresholdInPercentage,
+          deviationThresholdCoefficient
+        );
+
+        return checkUpdateConditions(
+          dapi.dataFeedValue.value,
+          dapi.dataFeedValue.timestamp,
+          newBeaconSetValue,
+          newBeaconSetTimestamp,
+          dapi.updateParameters.heartbeatInterval,
+          adjustedDeviationThresholdCoefficient
+        );
+      })
+      // Transform the batch and exclude beacons that cannot be updated
+      .map((dapiInfo) => ({
+        dapiInfo,
+        updatableBeacons: dapiInfo.decodedDataFeed.beacons
+          .filter(({ localDataIsStale }) => !localDataIsStale)
+          .map(({ beaconId, airnodeAddress, signedData, templateId }) => ({
+            beaconId,
+            airnodeAddress,
+            signedData,
+            templateId,
+          })),
+      }))
+      // Finally, filter out dapis that cannot be updated (eg. if we don't have signed data)
```

Comment:
```suggestion
      // Finally, filter out dapis that cannot be updated (e.g. if we don't have signed data)
```

Reply:
But can this case actually happen? I think if there are no updatable beacons then the dAPI will already be excluded in the first filter.

Reply:
mmm, I suspect it could maybe happen if the data on chain is updated between the application getting signed data and reading the on chain values? That would be a very rare scenario though. I can remove this check.